FWCore/ParameterSet/src/FillDescriptionFromPSet.cc

```diff
@@ -40,41 +41,34 @@ namespace {
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('E')] = &fillDescriptionFromParameter<edm::EventID>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('e')] =
-        &fillDescriptionFromParameter<std::vector<edm::EventID>>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('M')] =
-        &fillDescriptionFromParameter<edm::LuminosityBlockID>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('m')] =
+    s_findTheRightFunction[edm::k_int32] = &fillDescriptionFromParameter<int>;
+    s_findTheRightFunction[edm::k_vint32] = &fillDescriptionFromParameter<std::vector<int>>;
+    s_findTheRightFunction[edm::k_uint32] = &fillDescriptionFromParameter<unsigned>;
+    s_findTheRightFunction[edm::k_vuint32] = &fillDescriptionFromParameter<std::vector<unsigned>>;
+    s_findTheRightFunction[edm::k_int64] = &fillDescriptionFromParameter<long long>;
+    s_findTheRightFunction[edm::k_vint64] = &fillDescriptionFromParameter<std::vector<long long>>;
+    s_findTheRightFunction[edm::k_uint64] = &fillDescriptionFromParameter<unsigned long long>;
+    s_findTheRightFunction[edm::k_vuint64] = &fillDescriptionFromParameter<std::vector<unsigned long long>>;
+    s_findTheRightFunction[edm::k_double] = &fillDescriptionFromParameter<double>;
+    s_findTheRightFunction[edm::k_vdouble] = &fillDescriptionFromParameter<std::vector<double>>;
+    s_findTheRightFunction[edm::k_bool] = &fillDescriptionFromParameter<bool>;
+    s_findTheRightFunction[edm::k_stringRaw] = &fillDescriptionFromParameter<std::string>;
+    s_findTheRightFunction[edm::k_vstringHex] = &fillDescriptionFromParameter<std::vector<std::string>>;
+    s_findTheRightFunction[edm::k_EventID] = &fillDescriptionFromParameter<edm::EventID>;
+    s_findTheRightFunction[edm::k_VEventID] = &fillDescriptionFromParameter<std::vector<edm::EventID>>;
+    s_findTheRightFunction[edm::k_LuminosityBlockID] = &fillDescriptionFromParameter<edm::LuminosityBlockID>;
+    s_findTheRightFunction[edm::k_VLuminosityBlockID] =
         &fillDescriptionFromParameter<std::vector<edm::LuminosityBlockID>>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('t')] = &fillDescriptionFromParameter<edm::InputTag>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('v')] =
-        &fillDescriptionFromParameter<std::vector<edm::InputTag>>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('F')] = &fillDescriptionFromParameter<edm::FileInPath>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('A')] =
-        &fillDescriptionFromParameter<edm::LuminosityBlockRange>;
-    s_findTheRightFunction[static_cast<edm::ParameterTypes>('a')] =
+    s_findTheRightFunction[edm::k_InputTag] = &fillDescriptionFromParameter<edm::InputTag>;
+    s_findTheRightFunction[edm::k_VInputTag] = &fillDescriptionFromParameter<std::vector<edm::InputTag>>;
+    s_findTheRightFunction[edm::k_ESInputTag] = &fillDescriptionFromParameter<edm::ESInputTag>;
```

Comment:
These were the two missing entries.

The rest should just be a direct replacement of the char with the same valued enum.